Rishitha add new team form

RishithaMamidala opened the pull request:
# Description
(PRIORITY MEDIUM) : Phase 2: Create New Team (/bmdashboard/AddNewTeam) (WIP Rishitha)
Set Up Input Fields:
Create required input fields for:
Team Name (single-line input).
Additional Information (multiline text area).
Dropdown Menus:
Implement dropdowns for "Add Members" and "Add Main Tasks" with placeholder values.
Add Functionality:
Add an "ADD" button functionality for members and tasks:
On click, selected dropdown values appear as removable pills.
Allow dynamic addition and removal of members/tasks.
Pill Elements:
Create removable pills for "Currently assigned members" and "Currently assigned tasks" with an "X" button.
Ensure pills dynamically update upon addition or removal.
Form Validation:
Validate required fields:
Team Name must not be empty.
Optional: Display an error message for invalid inputs.
Button Actions:
Cancel: Reset all fields or navigate back to the previous page.
Submit: Collect all data and send it to the server or display it in the console for testing.
![Picture1](https://github.com/user-attachments/assets/349855e3-e717-47c0-be29-470c824a7371)

# Related PRS (if any):
This frontend PR is related to the development backend PR.

# Main changes explained:
Create file src/components/BMDashboard/Team/CreateNewTeam/CreateNewTeam.css for styling
Create file src/components/BMDashboard/Team/CreateNewTeam/CreateNewTeam.jsx to create new team form

# How to test:
check into current branch
do npm install and ... to run this PR locally
Clear site data/cache
log as admin user
go to http://localhost:3000/bmdashboard/AddNewTeam
verify the add team form
verify on submit the details are printed in the console

# Screenshots or videos of changes:

https://github.com/user-attachments/assets/bdc63e90-cd80-49c1-857a-eec0a44eec38


Note:
As of now no tasks are defined. So, task dropdown will be empty.

ziddigsm reviewed (changes requested):
No character limit for team name. No tasks in drop down to add main tasks. When I click on submit without adding a team member, it rightfully gives a message that I have to add a team member first. If I then add a team member, the error message to add a member doesnt disappear. Additionally,  a condition can be put to add atleast two team members before creating a team. The text alignment in cancel and submit buttons are also not aligned properly. Screenshots for the same are attached below.
<img width="1036" alt="Screenshot 2025-01-23 at 3 07 59 PM" src="https://github.com/user-attachments/assets/518868e8-7256-42e1-8024-a1c24d5a5e21" />
<img width="218" alt="Screenshot 2025-01-23 at 3 09 11 PM" src="https://github.com/user-attachments/assets/73ca1272-0f90-4701-82fc-7217c35796f1" />
<img width="439" alt="Screenshot 2025-01-23 at 3 09 34 PM" src="https://github.com/user-attachments/assets/dfaafbac-d385-48f9-bfa5-9b64223db4c1" />
<img width="502" alt="Screenshot 2025-01-23 at 3 13 00 PM" src="https://github.com/user-attachments/assets/5aa35428-7ad1-43f2-8afa-f0c19da795c8" />
<img width="512" alt="Screenshot 2025-01-23 at 3 14 04 PM" src="https://github.com/user-attachments/assets/fd4c80fc-db08-45cc-b444-037b59804ef5" />

srikanthpusthem reviewed (changes requested):
<img width="334" alt="image" src="https://github.com/user-attachments/assets/7552f824-7ef6-46b5-9993-dd4ddcacff9e" />

text in the button got over flown in smaller screens.

<img width="1710" alt="image" src="https://github.com/user-attachments/assets/8338c5e2-82c0-4b93-a537-609029f52775" />

when try adding same member it better if we prompt the user that the member is already added or it better not to show already added members in the dropdown

<img width="1710" alt="image" src="https://github.com/user-attachments/assets/b2ab923e-f4b1-43b5-b213-4a36c80c0f4c" />

no tasks are loaded in the select task drop down

RishithaMamidala commented:
> <img alt="image" width="334" src="https://private-user-images.githubusercontent.com/47666213/406563838-7552f824-7ef6-46b5-9993-dd4ddcacff9e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgwMTcxNDAsIm5iZiI6MTczODAxNjg0MCwicGF0aCI6Ii80NzY2NjIxMy80MDY1NjM4MzgtNzU1MmY4MjQtN2VmNi00NmI1LTk5OTMtZGQ0ZGRjYWNmZjllLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI3VDIyMjcyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJmOTUyYTc4ODRlMjA3ZjU1OTQ2ZDliZjIyZDU4ZDdiM2I2ZjUyYzQ3ZjZkZTMyYWY2NGZmMjhlY2ZmNDEwNGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1tceQGv5HU7N6DxVa8FGYyPYcHzyLlpk7CmhsLauauA">
> text in the button got over flown in smaller screens.
> 
> <img alt="image" width="1710" src="https://private-user-images.githubusercontent.com/47666213/406570991-8338c5e2-82c0-4b93-a537-609029f52775.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgwMTcxNDAsIm5iZiI6MTczODAxNjg0MCwicGF0aCI6Ii80NzY2NjIxMy80MDY1NzA5OTEtODMzOGM1ZTItODJjMC00YjkzLWE1MzctNjA5MDI5ZjUyNzc1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI3VDIyMjcyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAwNmU2MGMxODdmMmI1OWUwNzlmODZmNzY5NzUxYTYzOGZiZTYwNTRhMmJjMmMxZjlhZDAzYjU5NWI2MTYxZTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vDNVGrZafw7MXCf3YPsZcCMXXmt9fy96iGyKwbWUJyk">
> when try adding same member it better if we prompt the user that the member is already added or it better not to show already added members in the dropdown
> 
> <img alt="image" width="1710" src="https://private-user-images.githubusercontent.com/47666213/406572119-b2ab923e-f4b1-43b5-b213-4a36c80c0f4c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgwMTcxNDAsIm5iZiI6MTczODAxNjg0MCwicGF0aCI6Ii80NzY2NjIxMy80MDY1NzIxMTktYjJhYjkyM2UtZjRiMS00M2I1LWIyMTMtNGEzNmM4MGMwZjRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI3VDIyMjcyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEyYzk4NTFiOTEyYzEzNjViMWVhYjY4NDFmNDEyNmQ3MDE4YjQyMDNmMzI2ODljOGJkMzJiNjMzZDc4NjI2YzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hu4Re6cqra__VIC-OizX6pZnpZ6geTmJfVvgxprUfY4">
> no tasks are loaded in the select task drop down

@srikanthpusthem @ziddigsm 
I have implemented the following changes:
- Added character limit validation for the team name field.
- Corrected error display and ensured proper alignment of buttons across all screens.
But the task dropdown will be empty, because the tasks are not yet defined and it is out of the scope for this PR.
Please review the PR with these improvements and let me know if you have any further suggestions

ziddigsm reviewed (approved):
The changes requested were implemented and corrected.

<img width="577" alt="Screenshot 2025-01-27 at 7 15 30 PM" src="https://github.com/user-attachments/assets/40a98ac8-c1ca-447f-b353-71c54b42b940" />
<img width="516" alt="Screenshot 2025-01-27 at 7 16 30 PM" src="https://github.com/user-attachments/assets/836ab230-1610-4bad-b99f-2d839f87a5b1" />

fangedShadow reviewed (approved):
The form is resizing properly. the submit button and cancel button is working properly too and console logging the data. it is doing everything as expected. 
![Screenshot 2025-01-31 172201](https://github.com/user-attachments/assets/a6b47aae-c09d-4e2e-8e43-4a10379eeaa6)
![Screenshot 2025-01-31 172206](https://github.com/user-attachments/assets/1cf07eb2-0485-4b3c-87bb-36abbda6ef5c)
![Screenshot 2025-01-31 172829](https://github.com/user-attachments/assets/8c8611bc-1e27-4bc7-8fe4-6671ae53ef27)

mikelambo7 reviewed (approved):
Tested and verified that this works as expected

https://github.com/user-attachments/assets/3c60e1a5-b703-4853-87a1-33af14bff263

Srichand-Medagani reviewed (dismissed):
I have reviewed this PR and it working as expected. I am able to add a team without any errors in the console. 

https://github.com/user-attachments/assets/786ba582-918a-4b77-9dd6-0e2873974700

myeeli reviewed (dismissed):
I  have checked out the current PR and i could see the details in the console when submitting the team  form.
<img width="930" alt="PR 3064 - 1" src="https://github.com/user-attachments/assets/0fee8355-91bd-4fe4-8523-3a62f04b82ba" />
<img width="943" alt="PR 3064 - 2" src="https://github.com/user-attachments/assets/e65b5272-a3ac-4d11-bb7d-68022c622eae" />

linh2020 reviewed (dismissed):
Checked functionality:
- UI loads at /bmdashboard/AddNewTeam
- Form fields and validation work as expected
- Console logs correct data on Submit
- Pill system (add/remove) works for members
- Button layout and responsive view look good
- Dark mode is functional

Looks good to me.

![2025-03-29 09_24_22-HGN APP - Your Microsoft Edge - Microsoft​ Edge](https://github.com/user-attachments/assets/f0127dad-620f-4b81-bb35-8ca4eaf3797a)

![2025-03-29 09_27_09-HGN APP - Your Microsoft Edge - Microsoft​ Edge](https://github.com/user-attachments/assets/aeb5c1e7-d8a7-4bcd-ab9a-2fba5333fac4)

![2025-03-29 09_27_44-HGN APP - Your Microsoft Edge - Microsoft​ Edge](https://github.com/user-attachments/assets/f1f8b36b-812b-4099-bb3d-962d4cb75155)

one-community commented:
Thank you all, merging!
